### PR TITLE
Conversion Of Bot Task Module with Task Module Tab - Nodejs

### PR DESCRIPTION
**Pull request by Mohammed-MSFT**
Images and Readme Need to be update

### CODE REVIEW

**Comment by Pawan-MSFT** [comment]
Executed Build pipeline:
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/3821761962
image

File: samples/bot-task-module/nodejs/index.js [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 // Import required pckages
 const path = require('path');
+// const express = require("express");
```
Comment (Pawan-MSFT) [inline samples/bot-task-module/nodejs/index.js:8]: remove commented code

File: samples/bot-task-module/nodejs/pages/configure.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          microsoftTeams.pages.config.setConfig({
+            entityID: "Task",
+            contentUrl: `${window.location.origin}/taskmodule.html`,
+            suggestedTabName: "TaskModule",
+            websiteUrl: `${window.location.origin}/taskmodule.html`,
+          });
+          saveEvent.notifySuccess();
+        });
+      }
+    });
```
Comment (Pawan-MSFT) [inline samples/bot-task-module/nodejs/pages/configure.html:45]: This should be separate JS. for others also.

File: samples/bot-task-module/nodejs/pages/taskmodule.html [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+              microsoftTeams.tasks.startTask(taskInfo_1, submitHandler);
+              break;
+            default:
+              console.log("Unexpected button ID: " + this.id.toLowerCase());
+              return;
+          }
+          console.log("URL: " + taskInfo_1.url);
+        });
+      }
+    }
```
Comment (Pawan-MSFT) [inline samples/bot-task-module/nodejs/pages/taskmodule.html:272]: This should be separate JS.

File: samples/bot-task-module/nodejs/README.md [hunk 4]
```diff
@@ -77,7 +77,7 @@ In Azure portal, create a [Azure Bot resource](https://docs.microsoft.com/en-us/
 
 1) __*This step is specific to Teams.*__
```
Comment (Pawan-MSFT) [inline samples/bot-task-module/nodejs/README.md:78]: Gif for App-task-Module missing in readme.

**Review by Pawan-MSFT — COMMENTED** [review]
please fix comments

**Comment by ChetanSharma-msft** [comment]
Executed build pipeline:

File: samples/bot-task-module/nodejs/.env [hunk 5]
```diff
@@ -1,5 +1,5 @@
-MicrosoftAppType=
-MicrosoftAppId=
-MicrosoftAppPassword=
-MicrosoftAppTenantId=
-BaseUrl=
+MicrosoftAppType="<<MicrosoftAppType>>"
+MicrosoftAppId="<<MicrosoftAppId>>"
+MicrosoftAppPassword="<<MicrosoftAppPassword>>"
+MicrosoftAppTenantId="<<MicrosoftAppTenantId>>"
+BaseUrl="<<BaseUrl>>"
```
Comment (ChetanSharma-msft) [inline samples/bot-task-module/nodejs/.env:5]: Please remove extra added .gif file:

Comment (ChetanSharma-msft) [reply samples/bot-task-module/nodejs/.env:5]: image

File: samples/bot-task-module/nodejs/README.md [hunk 6]
```diff
@@ -77,7 +77,7 @@ In Azure portal, create a [Azure Bot resource](https://docs.microsoft.com/en-us/
...
     - **Edit** the `manifest.json` contained in the `appPackage` folder to replace your Microsoft App Id (that was created when you registered your bot earlier) *everywhere* you see the place holder string `<<YOUR-MICROSOFT-APP-ID>>` (depending on the scenario the Microsoft App Id may occur multiple times in the `manifest.json`)
-    - **Edit** the `manifest.json` for `validDomains` with base Url domain. E.g. if you are using ngrok it would be `https://1234.ngrok.io` then your domain-name will be `1234.ngrok.io`.
+    - **Edit** the `manifest.json` for `ValidDomain` with base Url domain. E.g. if you are using ngrok it would be `https://1234.ngrok.io` then your domain-name will be `1234.ngrok.io`.
```
Comment (ChetanSharma-msft) [inline samples/bot-task-module/nodejs/README.md:80]: `<<Valid-Domain>>`

File: samples/bot-task-module/nodejs/README.md [hunk 7]
```diff
@@ -106,10 +106,22 @@ You can interact with this bot by sending it a message. The bot will respond wit
...
 ## Deploy the bot to Azure
 
 To learn more about deploying a bot to Azure, see [Deploy your bot to Azure](https://aka.ms/azuredeployment) for a complete list of deployment instructions.
 
 ## Further reading
-
-- [How Microsoft Teams bots work](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-basics-teams?view=azure-bot-service-4.0&tabs=javascript)
\ No newline at end of file
+- [Use task modules in tabs](https://learn.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/task-modules/task-modules-tabs?tabs=teamsjs%2Cteamsjs2)
+- [Use task modules from bots](https://learn.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/task-modules/task-modules-bots?tabs=nodejs)
```
Comment (ChetanSharma-msft) [inline samples/bot-task-module/nodejs/README.md:127]: Remove en-us from these URL's.

File: samples/bot-task-module/nodejs/appPackage/manifest.json [hunk 8]
```diff
@@ -23,6 +23,27 @@
...
+      "canUpdateConfiguration": true,
+      "scopes": [
+        "team",
+        "groupchat"
+      ]
+    }
+  ],
+  "staticTabs": [
+    {
+      "entityId": "com.microsoft.teams.taskmoduletester",
```
Comment (ChetanSharma-msft) [inline samples/bot-task-module/nodejs/appPackage/manifest.json:38]: Give Proper entityId:
com.microsoft.teams.taskmoduletester

taskmodule.statictab

File: samples/bot-task-module/nodejs/pages/CustomForm.html [hunk 9]
```diff
@@ -1,4 +1,4 @@
-﻿<!DOCTYPE html>
+﻿﻿<!DOCTYPE html>
 <html lang="en">
 <head>
     <title>Microsoft Teams Task Module Tester - Custom Form</title>
```
Comment (ChetanSharma-msft) [inline samples/bot-task-module/nodejs/pages/CustomForm.html:4]: Microsoft Teams Task Module App

File: samples/bot-task-module/nodejs/pages/configure.html [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    <div>
+      <input type="radio" name="tabType" value="tasks" onclick="onClick()">Task Module<br>
+      <input type="radio" name="tabType" value="helloworld" onclick="onClick()"> Hello World
+    </div>
+    </p>
+  </form>
+  <script>
+    microsoftTeams.app.initialize().then(() => {
+      //get value of radio button
+      var radios = document.getElementsByName("tabType");
```
Comment (ChetanSharma-msft) [inline samples/bot-task-module/nodejs/pages/configure.html:21]: tabType

File: samples/bot-task-module/nodejs/pages/taskmodule.html [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+
+      var TaskModuleStrings = {
+        YouTubeTitle: "Microsoft Ignite 2018 Vision Keynote",
+        PowerAppTitle: "PowerApp: Asset Checkout",
+        CustomFormTitle: "Custom Form",
+        AdaptiveCardTitle: "Create a new job posting",
+        AdaptiveCardKitchenSinkTitle: "Adaptive Card: Inputs",
+        ActionSubmitResponseTitle: "Action.Submit Response",
+        YouTubeName: "YouTube",
+        PowerAppName: "PowerApp",
```
Comment (ChetanSharma-msft) [inline samples/bot-task-module/nodejs/pages/taskmodule.html:71]: Add proper alignment and spacing

File: samples/bot-task-module/nodejs/pages/taskmodule.html [hunk 12]
```diff
@@ -0,0 +1,274 @@
...
+        var btn = taskModuleButtons_1[_i];
+        btn.addEventListener("click", function () {
+          // Hide customFormResults, adaptiveResults
+          var customFormResults = document.getElementById("customFormResults");
+          customFormResults.style.display = "none";
+          var adaptiveResults = document.getElementById("adaptiveResults");
+          adaptiveResults.style.display = "none";
+          taskInfo_1.url = "".concat((0, appRoot)(), "/").concat(this.id.toLowerCase(), ".html");
+          // Define default submitHandler()
+          var submitHandler = function (err, result) { console.log("Err: ".concat(err, "; Result:  + ").concat(result)); };
```
Comment (ChetanSharma-msft) [inline samples/bot-task-module/nodejs/pages/taskmodule.html:229]: Alignment and spacing is not correct.

File: samples/bot-task-module/nodejs/pages/taskmodule.html [hunk 13]
```diff
@@ -0,0 +1,274 @@
...
+              taskInfo_1.title = TaskModuleStrings.AdaptiveCardTitle;
+              taskInfo_1.url = "";
+              taskInfo_1.size = { height: TaskModuleSizes.powerapp.height, width: TaskModuleSizes.powerapp.width };
+              taskInfo_1.card = adaptiveCard_template;
+              var resultsElement = document.getElementById("adaptiveResults");
+              submitHandler = (err, result) => {
+                console.log(`Submit handler - err: ${err}`);
+                console.log(
+                  "Result = " + JSON.stringify(result) + "\nError = " + JSON.stringify(err)
+                );
```
Comment (ChetanSharma-msft) [inline samples/bot-task-module/nodejs/pages/taskmodule.html:261]: Alignment and spacing

File: samples/bot-task-module/nodejs/pages/taskmodule.html [hunk 14]
```diff
@@ -0,0 +1,274 @@
...
+              };
+              microsoftTeams.tasks.startTask(taskInfo_1, submitHandler);
```
Comment (ChetanSharma-msft) [inline samples/bot-task-module/nodejs/pages/taskmodule.html:263]: one line space.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments.

**Comment by ChetanSharma-msft** [comment]
@Mohammed-MSFT - Please execute build pipeline once comments fixes are done.

**Review by Pawan-MSFT — COMMENTED** [review]
Still showing comments please fix these.

File: samples/bot-task-module/nodejs/README.md (hunk 6 above)
Comment (Mohammed-MSFT) [reply samples/bot-task-module/nodejs/README.md:80]: fixed

File: samples/bot-task-module/nodejs/README.md (hunk 7 above)
Comment (Mohammed-MSFT) [reply samples/bot-task-module/nodejs/README.md:127]: fixed

File: samples/bot-task-module/nodejs/appPackage/manifest.json (hunk 8 above)
Comment (Mohammed-MSFT) [reply samples/bot-task-module/nodejs/appPackage/manifest.json:38]: fixed

File: samples/bot-task-module/nodejs/pages/CustomForm.html (hunk 9 above)
Comment (Mohammed-MSFT) [reply samples/bot-task-module/nodejs/pages/CustomForm.html:4]: fixed

File: samples/bot-task-module/nodejs/pages/configure.html (hunk 10 above)
Comment (Mohammed-MSFT) [reply samples/bot-task-module/nodejs/pages/configure.html:21]: fixed

File: samples/bot-task-module/nodejs/pages/taskmodule.html (hunk 11 above)
Comment (Mohammed-MSFT) [reply samples/bot-task-module/nodejs/pages/taskmodule.html:71]: fixed

File: samples/bot-task-module/nodejs/pages/taskmodule.html (hunk 12 above)
Comment (Mohammed-MSFT) [reply samples/bot-task-module/nodejs/pages/taskmodule.html:229]: fixed

File: samples/bot-task-module/nodejs/pages/taskmodule.html (hunk 13 above)
Comment (Mohammed-MSFT) [reply samples/bot-task-module/nodejs/pages/taskmodule.html:261]: fixed

File: samples/bot-task-module/nodejs/pages/taskmodule.html (hunk 14 above)
Comment (Mohammed-MSFT) [reply samples/bot-task-module/nodejs/pages/taskmodule.html:263]: fixed

**Comment by ChetanSharma-msft** [comment]
Executed build pipeline: https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/3823889956

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Comment by Mohammed-MSFT** [comment]
Build Pipeline Executed : https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/3823936899
image

**Review by Wajeed-msft — APPROVED** [review]

**Comment by Mohammed-MSFT** [comment]
Executed Build Pipeline After Merge Conflicts Resolved : https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/3829237176

image

**Pull request merged**